File: app/src/main/java/hr/bpervan/novaeva/activities/MolitvenikActivity.java
package hr.bpervan.novaeva.activities;

import hr.bpervan.novaeva.NovaEvaApp;
import hr.bpervan.novaeva.main.R;
import hr.bpervan.novaeva.adapters.MolitvenikAdapter;

import java.util.ArrayList;
import java.util.List;

import android.app.AlertDialog;
import android.app.ListActivity;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.view.Menu;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.EditText;
import android.widget.ListView;

/*import com.google.analytics.tracking.android.EasyTracker;
import com.google.analytics.tracking.android.GoogleAnalytics;
import com.google.analytics.tracking.android.Tracker;*/


public class MolitvenikActivity extends ListActivity implements OnClickListener{
	/*private Tracker mGaTracker;
	private GoogleAnalytics mGaInstance;*/
	
	private ListView mainListView;
	private MolitvenikAdapter molitvenikAdapter;
	private List<String> mainContentList;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_molitvenik);
		
		/*mGaInstance = GoogleAnalytics.getInstance(this);
		mGaTracker = mGaInstance.getTracker("UA-40344870-1");*/
		
		initUI();
				
		this.setTitle("Molitvenik");
		
		mainContentList = new ArrayList<String>();
		
		mainContentList.add("Često tražene molitve");
		mainContentList.add("0. Uvod");
		mainContentList.add("1. Obrasci vjere");
		mainContentList.add("2. Osnovne molitve");
		mainContentList.add("3. Svagdanje jutarnje molitve");
		mainContentList.add("4. Svagdanje večernje molitve");
		mainContentList.add("5. Prigodne molitve");
		mainContentList.add("6. Molitve mladih");
		mainContentList.add("7. Molitve u kušnji i napasti");
		mainContentList.add("8. Molitve za obitelj i roditelje");
		mainContentList.add("9. Molitve za bolesne i umiruće");
		mainContentList.add("10. Molitve po posebnim nakanama");
		mainContentList.add("11. Molitve svetih i velikih ljudi");
		mainContentList.add("12. Kratke molitve i zazivi");
		mainContentList.add("13. Molitve Duhu Svetome");
		mainContentList.add("14. Euharistijska pobožnost");
		mainContentList.add("15. Pomirenje");
		mainContentList.add("16. Pobožnost križnog puta");
		mainContentList.add("17. Deventica i krunica Božanskom milosrđu");
		mainContentList.add("18. Molitve Blaženoj Djevici Mariji");
		mainContentList.add("19. Salezijanske molitve");
		mainContentList.add("20. Molitve mladih");
		mainContentList.add("21. Molitve svetima");
		mainContentList.add("22. Lectio Divina");
		mainContentList.add("23. Moliti igrajući pred Gospodinom");
		
		molitvenikAdapter = new MolitvenikAdapter(this,mainContentList);

		mainListView.setAdapter(molitvenikAdapter);
		
		mainListView.setOnItemClickListener(new OnItemClickListener(){
			@Override
			public void onItemClick(AdapterView<?> arg0, View arg1, int position,
					long id) {
				
				//mGaTracker.sendEvent("Molitvenik", "OtvorenaMolitva", String.valueOf(position), null);
				Intent intent=new Intent(MolitvenikActivity.this,MolitvenikDetaljiActivity.class);
				intent.putExtra("id",String.valueOf(position));
				intent.putExtra("naslov", mainContentList.get(position));
				startActivity(intent);			
			}				
		});
	}
	
	@Override
	public void onConfigurationChanged(Configuration newConfig){
		super.onConfigurationChanged(newConfig);
		setContentView(R.layout.activity_molitvenik);
		
		initUI();
		
		mainListView.setAdapter(molitvenikAdapter);
		mainListView.setOnItemClickListener(new OnItemClickListener(){
			@Override
			public void onItemClick(AdapterView<?> arg0, View arg1, int position,
					long id) {
				
				//mGaTracker.sendEvent("Molitvenik", "OtvorenaMolitva", String.valueOf(position), null);
				Intent intent=new Intent(MolitvenikActivity.this,MolitvenikDetaljiActivity.class);
				intent.putExtra("id",String.valueOf(position));
				intent.putExtra("naslov", mainContentList.get(position));
				startActivity(intent);			
			}				
		});
	}
	
	private void initUI(){
		mainListView = getListView();
		mainListView.setClickable(true);

		View fakeActionBar = findViewById(R.id.fakeActionBar);
		fakeActionBar.findViewById(R.id.btnBack).setOnClickListener(this);
		fakeActionBar.findViewById(R.id.btnSearch).setOnClickListener(this);
		fakeActionBar.findViewById(R.id.btnHome).setOnClickListener(this);
	}
	
	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		//getMenuInflater().inflate(R.menu.activity_molitvenik, menu);
		return true;
	}
	
	@Override
	public void onClick(View v) {
		switch(v.getId()){
		case R.id.btnSearch:
			showSearchPopup();
			break;
		case R.id.btnHome:
			NovaEvaApp.Companion.goHome(this);
			break;
		case R.id.btnBack:
			onBackPressed();
			break;
		}
	}
	
	private void showSearchPopup(){
		AlertDialog.Builder search = new AlertDialog.Builder(this);
		search.setTitle("Pretraga");
		
		final EditText et = new EditText(this);
		search.setView(et);
				
		search.setPositiveButton("Pretraži", new DialogInterface.OnClickListener() {
			@Override
			public void onClick(DialogInterface dialog, int which) {
				String search = et.getText().toString();
                NovaEvaApp.Companion.goSearch(search, MolitvenikActivity.this);
			}
		});
		search.setNegativeButton("Odustani", new DialogInterface.OnClickListener() {
			@Override
			public void onClick(DialogInterface dialog, int whichButton) {}
			});
		search.show();
	}

}

File: app/src/main/java/hr/bpervan/novaeva/activities/MolitvenikActivity.kt
package hr.bpervan.novaeva.activities

import android.app.AlertDialog
import android.app.ListActivity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.View.OnClickListener
import android.widget.AdapterView.OnItemClickListener
import android.widget.EditText
import android.widget.ListView
import hr.bpervan.novaeva.NovaEvaApp
import hr.bpervan.novaeva.adapters.MolitvenikAdapter
import hr.bpervan.novaeva.main.R
import kotlinx.android.synthetic.main.activity_molitvenik.*
import kotlinx.android.synthetic.main.simple_fake_action_bar.view.*

/*import com.google.analytics.tracking.android.EasyTracker;
import com.google.analytics.tracking.android.GoogleAnalytics;
import com.google.analytics.tracking.android.Tracker;*/

class MolitvenikActivity : ListActivity(), OnClickListener {
    /*private Tracker mGaTracker;
	private GoogleAnalytics mGaInstance;*/

    private lateinit var mainListView: ListView
    private lateinit var mainContentList: List<String>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_molitvenik)

        /*mGaInstance = GoogleAnalytics.getInstance(this);
		mGaTracker = mGaInstance.getTracker("UA-40344870-1");*/

        initUI()

        mainContentList = listOf(
                "Često tražene molitve",
                "0. Uvod",
                "1. Obrasci vjere",
                "2. Osnovne molitve",
                "3. Svagdanje jutarnje molitve",
                "4. Svagdanje večernje molitve",
                "5. Prigodne molitve",
                "6. Molitve mladih",
                "7. Molitve u kušnji i napasti",
                "8. Molitve za obitelj i roditelje",
                "9. Molitve za bolesne i umiruće",
                "10. Molitve po posebnim nakanama",
                "11. Molitve svetih i velikih ljudi",
                "12. Kratke molitve i zazivi",
                "13. Molitve Duhu Svetome",
                "14. Euharistijska pobožnost",
                "15. Pomirenje",
                "16. Pobožnost križnog puta",
                "17. Deventica i krunica Božanskom milosrđu",
                "18. Molitve Blaženoj Djevici Mariji",
                "19. Salezijanske molitve",
                "20. Molitve mladih",
                "21. Molitve svetima",
                "22. Lectio Divina",
                "23. Moliti igrajući pred Gospodinom")


        mainListView.adapter = MolitvenikAdapter(this, mainContentList)

        mainListView.onItemClickListener = OnItemClickListener { arg0, arg1, position, id ->
            //mGaTracker.sendEvent("Molitvenik", "OtvorenaMolitva", String.valueOf(position), null);
            val intent = Intent(this@MolitvenikActivity, MolitvenikDetaljiActivity::class.java)
            intent.putExtra("id", position.toString())
            intent.putExtra("naslov", mainContentList[position])
            startActivity(intent)
        }
    }

    private fun initUI() {

        this.title = "Molitvenik"

        mainListView = listView
        mainListView.isClickable = true

        fakeActionBar.btnBack.setOnClickListener(this)
        fakeActionBar.btnSearch.setOnClickListener(this)
        fakeActionBar.btnHome.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btnSearch -> showSearchPopup()
            R.id.btnHome -> NovaEvaApp.goHome(this)
            R.id.btnBack -> onBackPressed()
        }
    }

    private fun showSearchPopup() {
        val searchBuilder = AlertDialog.Builder(this)
        searchBuilder.setTitle("Pretraga")

        val et = EditText(this)
        searchBuilder.setView(et)

        searchBuilder.setPositiveButton("Pretraži") { _, _ ->
            val search = et.text.toString()
            NovaEvaApp.goSearch(search, this@MolitvenikActivity)
        }
        searchBuilder.setNegativeButton("Odustani") { _, _ -> }
        searchBuilder.show()
    }

}
